### PR TITLE
working instance of geogig

### PR DESCRIPTION
**Pull request by NyakudyaA**
Working instance of geogig as a service

### CODE REVIEW

File: 71-apt-cacher-ng [hunk 1]
```diff
@@ -1 +1 @@
-Acquire::http { Proxy "http://192.168.2.3:3142"; };
+Acquire::http { Proxy "http://192.168.10.200:3142"; };
```
Comment (timlinux) [inline 71-apt-cacher-ng:1]: This should be commented out in the public repo

File: Dockerfile [hunk 2]
```diff
@@ -1,47 +1,36 @@
-#--------- Generic stuff all our Dockerfiles should start with so we get caching ------------
-FROM ubuntu:trusty
-MAINTAINER Tim Sutton<tim@linfiniti.com>
+FROM ubuntu:14.04
+MAINTAINER admire@afrispatial.co.za
```
Comment (timlinux) [inline Dockerfile:2]: kartoza right?

File: Dockerfile [hunk 3]
```diff
@@ -1,47 +1,36 @@
...
 
-RUN  export DEBIAN_FRONTEND=noninteractive
-ENV  DEBIAN_FRONTEND noninteractive
-RUN  dpkg-divert --local --rename --add /sbin/initctl
-#RUN  ln -s /bin/true /sbin/initctl
 
-# Use local cached debs from host (saves your bandwidth!)
-# Change ip below to that of your apt-cacher-ng host
-ADD 71-apt-cacher-ng /etc/apt/apt.conf.d/71-apt-cacher-ng
+#ADD 71-apt-cacher-ng /etc/apt/apt.conf.d/71-apt-cacher-ng
```
Comment (timlinux) [inline Dockerfile]: No dont comment out here, comment out in 71-apt-cacher- file rather

File: Dockerfile [hunk 4]
```diff
@@ -1,47 +1,36 @@
...
-
-# Next line a workaround for https://github.com/dotcloud/docker/issues/963
-RUN apt-get install -y --no-install-recommends openjdk-7-jdk
+RUN apt-get -y install  default-jdk
+RUN apt-get  install -y wget unzip daemontools
+RUN mkdir -p  /etc/service
+RUN mkdir -p  /etc/service/geogig_serve
+WORKDIR /etc/service/geogig_serve
+
+RUN touch run
```
Comment (timlinux) [inline Dockerfile]: not needed unless you are trying to edit an exiting file

File: Dockerfile [hunk 5]
```diff
@@ -1,47 +1,36 @@
...
+RUN echo "#!/bin/bash" >>run
```
Comment (timlinux) [inline Dockerfile:22]: first redirect can be > (single >) which will create the file as an empty file

File: README.md [hunk 6]
```diff
@@ -7,7 +7,7 @@ host:
 sudo apt-get install docker
```
Comment (timlinux) [inline README.md]: Our primary usage instruction should be:

```
docker pull kartoza/geogig
sudo docker run --name="geogig" -p 38080:8182 -d kartoza/geogig
```

and the rest shoul be for the hard core.

File: README.md [hunk 7]
```diff
@@ -23,15 +23,13 @@ maven build grabs a lot of jars.
 After it is installed run a container:
 
 ```bash
-sudo docker run --name="geogit" -p 38080:8080 -d -t kartoza/geogit /start.sh
+sudo docker run --name="geogig" -p 38080:8182 -i -d -t kartoza/geogig
```
Comment (timlinux) [inline README.md]: Its running as a daemon right? So no -i -t in this line...

File: build.sh [hunk 8]
```diff
@@ -1 +1,3 @@
-docker build -t kartoza/geogit:1.0 .
+#bin/bash
+git clone http://github.com/boundlessgeo/GeoGig.git
```
Comment (timlinux) [inline build.sh]: We should have an alternative logic path in the docker file that checks out in the build process in the build context if the geogig dir is not present. Above build wont workon the docker build service.

File: setup.sh [hunk 9]
```diff
@@ -1,14 +1,18 @@
 # Once-off tasks to run when setting up the container
 # Tim Sutton, April 24, 2014
 # Install GeoGit
-cd /
-git clone http://github.com/boundlessgeo/GeoGit.git
-cd GeoGit/src/parent
+#cd /
+#git clone http://github.com/boundlessgeo/GeoGig.git
+#cd GeoGig/src/parent
+cd /GeoGig/src/parent
```
Comment (timlinux) [inline setup.sh:5]: This wont work if the geogig is not checked out already which will fail on the docker build service

File: Dockerfile (hunk 5 above)
Comment (NyakudyaA) [inline Dockerfile:22]: using the > to create an empty file results in the container not working at all.

**Comment by NyakudyaA** [comment]
New pull request will cover all issues and comments made by Tim

**Pull request closed without merging**